### ISSUE TITLE
[Dev setup] godep update does not work on OS X

### ISSUE THREAD

**Issue by ppadala** [docker/machine#3337]
I am trying to update an existing vendored library (in this case github.com/mitchellh/mapstructure). Usual process is

1. make dep-restore
2. go get -u <repo> # get latest
3. godep update
4. make dep-save

Step 3 fails on Mac with 

```
Pradeeps-MacBook-Air-2:machine ppadala$ godep update github.com/mitchellh/mapstructure
godep: no buildable Go source files in /Users/ppadala/gomachine/src/golang.org/x/sys/windows/registry
godep: error loading dependencies
```

I suspect some versioning problems, but same steps on Linux work fine.

**Comment by nathanleclaire** [comment]
Yeah, this is messy right now.  It'd probably be well worth my while to take some time and try and clean it up if possible.  Thanks for the issue, I'll look into it.  Weird that it works on Linux but not OSX.

**Comment by ahmetalpbalkan** [comment]
yeah I ran into this as well. godep is a mess. if you want to update a dependency:

1. `make dep-restore`
2. refactor code and comment out anything that uses that package (i.e. take dat pkg out of build somehow, directly or indirectly)
3. `make dep-save`
4. `go get pkgname`
4. undo commenting out (i.e. take it back to build)
5. `make dep-save`
6. should be updated (check Godeps.json)

note that I haven't used `godep update` at all.

**Comment by runseb** [comment]
@nathanleclaire I am running into somewhat similar issues. What is the proper way to update a package that's needed for a driver. Godeps or vendor/

**Comment by nathanleclaire** [comment]
`vendor/`, but I think it gets updated through `godep`

**Comment by nogoegst** [comment]
See #4395 for migration to golang/dep.

**Comment by nogoegst** [comment]
Should be closed due to #4395 and timeout.
